pandas/tests/extension_arrays/base.py

```diff
@@ -405,3 +405,35 @@ def test_value_counts(self, all_data, dropna):
         expected = pd.Series(other).value_counts(dropna=dropna).sort_index()
 
         tm.assert_series_equal(result, expected)
+
+    def test_count(self, data_missing):
+        df = pd.DataFrame({"A": data_missing})
+        result = df.count(axis='columns')
+        expected = pd.Series([0, 1])
+        tm.assert_series_equal(result, expected)
+
+    def test_dropna_series(self, data_missing):
+        ser = pd.Series(data_missing)
+        result = ser.dropna()
+        expected = ser.iloc[[1]]
+        tm.assert_series_equal(result, expected)
```

Comment:
Can you add a check here that `result` of dropna is still ExtensionArray based? (not materialized) Because `assert_series_equal` will not catch that. 
Maybe we should add a helper function to do that (as you want to ensure that in almost all operations.

Reply:
Or maybe better, fix `assert_series_equal` :-)
So adapt it to also check the array type (extension type or not), and not only it's values ?

Reply:
Hm, I think we (implicitly) check that already in the `check_dtype`

```
In [16]: tm.assert_series_equal(left, right)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-16-70cf126e5879> in <module>()
----> 1 tm.assert_series_equal(left, right)
...

AssertionError: Attributes are different

Attribute "dtype" are different
[left]:  decimal
[right]: object
```

I suppose someone could make an extension array that uses 'object' for the name, but that's just a bad idea :)

Reply:
Ah, sorry, you are right! I was only checking the assert of the values that happens inside assert_series_equal, and that one does not care (as numpy coerce both inputs to ndarray)